lib/nexpose/scan_template.rb

```diff
@@ -176,6 +176,56 @@ def host_threads=(threads)
       host_threads.text = threads.to_s
     end
 
+    # Enable/disable IP stack fingerprinting
+    # @param [Boolean] enable or disable IP stack fingerprinting
+    def enable_ip_stack_fingerprinting(enable)
+      ns = REXML::XPath.first(@xml, 'ScanTemplate/Plugins/Plugin[@name="java/NetworkScanners"]')
+      param = REXML::XPath.first(ns, './param[@name="ipFingerprintEnabled"]')
+      unless param
+        param = REXML::Element.new('param')
+        param.add_attribute('name', 'ipFingerprintEnabled')
+        ns.add_element(param)
+      end
+      param.text = (enable ? 1 : 0)
+    end
+
+    # Enable/diisable ICMP device discovery
+    # @param [Boolean] enable or disable ICMP device discovery
+    def enable_icmp_discovery=(enable)
+      icmp = REXML::XPath.first(@xml, 'ScanTemplate/DeviceDiscovery/CheckHosts/icmpHostCheck')
+      icmp.attributes['enabled'] = (enable ? 1 : 0)
+    end
+
+    # Add custom TCP ports to scan for device discovery
+    # @param [Array] ports to scan for device discovery
+    def tcp_discovery_ports=(ports)
+      tcp = REXML::XPath.first(@xml, 'ScanTemplate/DeviceDiscovery/CheckHosts/TCPHostCheck')
+      enable_tcp_discovery = true
+      REXML::XPath.first(tcp, './portList').text = ports.join(",")
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.